flowmachine/flowmachine/features/subscriber/active_subscribers.py

```diff
@@ -0,0 +1,73 @@
+from datetime import date
+from typing import List, Union, Optional
+from functools import reduce
+
+from flowmachine.core.mixins.exposed_datetime_mixin import ExposedDatetimeMixin
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.total_active_periods import (
+    TotalActivePeriodsSubscriber,
+)
+from flowmachine.features.utilities.events_tables_union import EventsTablesUnion
+from dateutil.rrule import rrule, DAILY
+
+
+class ActiveSubscribers(ExposedDatetimeMixin, Query):
+    """Returns a list of subscribers seen at least `active_days` between `start_date` and `end_date`,
+    where 'active' is at least `active_hours` call-hours active
+    """
+
+    def __init__(
+        self,
+        start_date: Union[date, str],
+        end_date: Union[date, str],
+        active_hours: int,
+        active_days: int,
+        subscriber_id: str = "msisdn",
+        events_tables: Optional[List[str]] = None,
+        subscriber_subset=None,
+    ):
+        self.start_date = start_date
+        self.end_date = end_date
+        self.active_hours = active_hours
+        self.sub_id_column = subscriber_id
```

Comment:
Parameters that are just getting passed down to the sub-queries (sub_id_column and events_tables here) don't necessarily need to be stored as attributes, if they're not directly used within this query.

I'd also say it would be preferable to name these parameters 'subscriber_identifier' and 'tables', for consistency with most (admittedly not all) other queries that accept these parameters. (Side note: I think the fact that we have so many queries that accept the same set of parameters, which all get passed down to an underlying `EventsTablesUnion` query, is a sign that we could do with some refactoring. But that's something for another time, and will be easier to refactor if the names of these parameters are reasonably consistent to start with).